packages/nimble-components/src/table/specs/table-column-width-hld.md

```diff
@@ -0,0 +1,170 @@
+
+### Expected Behavior
+
+The behavior that has been prescribed for column sizing is as follows:
+
+- Columns should be able to be configured to either maintain a fixed width, or grow proportionally with the table such as when the window resizes causing the table width to increase. Tables can consist of columns that are configured as a mixture of the two modes.
+- If a user drags a divider between two columns to the right, then the column on the left will grow larger, and the column on the right will grow smaller by the same pixel amount. Sub-behaviors to this are:
+    - If a shrinking column has reached its minimum pixel size, then the next column in the direction of the sizing action will be affected up to the final column in a given direction.
+    - A sizing action to the left will ultimately stop having an effect when the left-most column reaches its minimum size.
+    - A sizing action to the right that would ultimately result in the final right column reaching its minimum size (all columns still within current table width) would then begin to push columns out of the table width resulting in a horizontal scrollbar.
+- Columns can be configured to not allow a user to interactively size them
+    - The implicit behavior present based on the behaviors described above, is that in a sizing action that cascades to a column configured to not be resized is that the column won't be resized towards a minimum size, and the cascade will effectively "skip" this column.
+
+### Out of Scope
+
+There are some column sizing behaviors that we will ultimately expect to support, but the APIs presented here are not meant to address:
+- Auto-resizing: We will not describe how we intend to support the use-case of having a column auto-size to its contents
+- Different interactive sizing modes: While the APIs described in this HLD do not inherently prescribe to a particular interactive sizing behavior, it's worth saying that in order to support multiple sizing modes, there will likely be additional APIs required that this HLD does not address.
+- Mechanisms related to accessibility-centric interactive column sizing (if there are such mechanisms). One possible example is allowing a user to size a column by way of the keyboard, instead of using a mouse. Ultimately, such a scenario is not in conflict with the API presented here, nor the mouse-based approach we know we will require, and can thus be handled separately, if ever.
+
+### API
+
+To accomodate the various sizing modes of a column, in addition to the other necessary behaviors, we can add the following properties to `TableColumn`:
+
+```
+class TableColumn {
+    // when set 'currentFractionalWidth' should be ignored
+    @observable
+    public currentPixelWidth: number | null = null;
+
+    @observable
+    public currentFractionalWidth = 1;
+
+    @observable
+    public currentMinWidth = 88; // the minimum size according to the design doc
+
+    @observable
+    public currentDisableResize = false;
```

Comment:
Technically, I'm not sure we would call it internal. Primarily because the proposed mixin class would be directly leveraging those APIs (though we could argue that it's internal to Nimble). The less convincing reason is that it's conceivable (though likely undesirable) for a client to implement their own mixin requiring the same access to those APIs.

Despite all of this, I'm not sure how important it is to capture the documented visibility of the properties in this HLD. It certainly is relevant to the ultimate implementation.

Reply:
I think it's expected that specs / HLD document our public API. Not stating they are internal means they are end user facing. Eng user's should not be expected to mixin behaviors. Plugin authors can be expected to use mixins and be aware of internal apis they need to use for plugin integration.

Reply:
Marking them internal impacts the generated type definitions from typescript and therefore the public api surface area.